### ISSUE TITLE
Possible misuse of `aria-label` on `option` elements

### ISSUE THREAD

**Issue by hrvojegolcic** [validator/validator#1305]
Hi,

I am writing here because I got a a following message in a validator:

> Warning: Possible misuse of aria-label. (If you disagree with this warning, file an issue report or send e-mail to www-validator@w3.org.)

The code snippet being validated:

```
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>
<label for="search_type">Filter</label>
<select class="form-control" id="search_type">
  <option value="0">Alle</option>
  <option value="1">Projekte</option>
  <option value="2">Vakanzen</option>
  <option value="3">Stellenanzeigen</option>
  <optgroup label="Bewerber">
    <option value="4" aria-label="Bewerber Alle">Alle</option>
    <option value="5" aria-label="Bewerber Familienname">Familienname</option>
    <option value="6" aria-label="Bewerber Kommentare">Kommentare</option>
    <option value="7" aria-label="Bewerber Anhänge">Anhänge</option>
  </optgroup>
</select>
</body>
</html>
```

The thing is, `aria-label` is not illegal on `option`, so it's not a validation issue. And in this case specifically, it was crucial, because from my testing I figured out that optgroup labelling are simply not supported by screen readers JAWS and NVDA hence the information was hidden from the users. The ARIA "trick" does not look like a bad use to work around it. In fact it's the only solution I found. Suggestions? Or maybe not to report this issue on `option` elements until we see a better support from screen readers.

**Comment by sideshowbarker** [comment]
@scottaohara @pkra

**Comment by pkra** [comment]
Looks like a bug. The option element is mapped to role=option as per https://w3c.github.io/html-aam/#el-option, aria-label is allowed on role=option as per https://w3c.github.io/aria/#option, and no restrictions from html-aria https://w3c.github.io/html-aria/#el-option. AT seems fine with it, too.

**Comment by sideshowbarker** [comment]
Fixed by fb1bf7be70809c950595b6d7cd98b8aa51ed2656

See https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bcharset%3Dutf-8%3Bbase64%2CPCFET0NUWVBFIGh0bWw%2BDQo8aHRtbCBsYW5nPSIiPg0KPGhlYWQ%2BDQo8dGl0bGU%2BVGVzdDwvdGl0bGU%2BDQo8L2hlYWQ%2BDQo8Ym9keT4NCjxsYWJlbCBmb3I9InNlYXJjaF90eXBlIj5GaWx0ZXI8L2xhYmVsPg0KPHNlbGVjdCBjbGFzcz0iZm9ybS1jb250cm9sIiBpZD0ic2VhcmNoX3R5cGUiPg0KICA8b3B0aW9uIHZhbHVlPSIwIj5BbGxlPC9vcHRpb24%2BDQogIDxvcHRpb24gdmFsdWU9IjEiPlByb2pla3RlPC9vcHRpb24%2BDQogIDxvcHRpb24gdmFsdWU9IjIiPlZha2FuemVuPC9vcHRpb24%2BDQogIDxvcHRpb24gdmFsdWU9IjMiPlN0ZWxsZW5hbnplaWdlbjwvb3B0aW9uPg0KICA8b3B0Z3JvdXAgbGFiZWw9IkJld2VyYmVyIj4NCiAgICA8b3B0aW9uIHZhbHVlPSI0IiBhcmlhLWxhYmVsPSJCZXdlcmJlciBBbGxlIj5BbGxlPC9vcHRpb24%2BDQogICAgPG9wdGlvbiB2YWx1ZT0iNSIgYXJpYS1sYWJlbD0iQmV3ZXJiZXIgRmFtaWxpZW5uYW1lIj5GYW1pbGllbm5hbWU8L29wdGlvbj4NCiAgICA8b3B0aW9uIHZhbHVlPSI2IiBhcmlhLWxhYmVsPSJCZXdlcmJlciBLb21tZW50YXJlIj5Lb21tZW50YXJlPC9vcHRpb24%2BDQogICAgPG9wdGlvbiB2YWx1ZT0iNyIgYXJpYS1sYWJlbD0iQmV3ZXJiZXIgQW5ow6RuZ2UiPkFuaMOkbmdlPC9vcHRpb24%2BDQogIDwvb3B0Z3JvdXA%2BDQo8L3NlbGVjdD4NCjwvYm9keT4NCjwvaHRtbD4%3D
